tedge/src/cli/connect/command.rs

```diff
@@ -176,7 +184,40 @@ where
 // the check can finish in the second try as there is no error response in the first try.
 
 #[tokio::main]
-async fn check_connection_c8y(port: u16) -> Result<(), ConnectError> {
+async fn check_connection_c8y(port: u16) -> Result<DeviceStatus, ConnectError> {
+    for i in 0..2 {
+        print!("Try {} / 2: Sending a message to Cumulocity. ", i + 1,);
```

Comment:
Isn't it misleading to send the message when the device is not created at all.

Reply:
That statement is just about sending a message and doesn't say that it's a device creation request in the background. So, not sure why you think it's misleading. We will have to send some message to the cloud to check if the device exists or not, right? So the words "sending a message" sounds appropriate to me.

Reply:
I was thinking more from the code perspective, not the end-user. Also, the check_connection just should check the connection not create the device. By the time we call check_connection, the device must be created already. I feel it works properly in the case of azure.

Reply:
Unfortunately, Cumulocity doesn't have a SmartREST message that lets us check if the device is indeed connected to a device twin on the cloud. That's why we've used this hacky mechanism of "create a device and see if there's a response" mechanism.

Reply:
To avoid the confusion while reading code, I've renamed the `check_connection_c8y` method to `check_device_status_c8y` and the same for Azure.